### PR TITLE
chore: add CodeQL Analysis to Repo

### PR DESCRIPTION
**Pull request by fubuloubu**

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  schedule:
+    - cron: '23 17 * * 5'
```
Comment (unparalleled-js) [inline .github/workflows/codeql.yml:9]: Does this need to run on a schedule?

**Review by unparalleled-js — COMMENTED** [review]
It is showing one alert in `test_project.py`, do we want to also address that?

**Review by unparalleled-js — DISMISSED** [review]

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (fubuloubu) [reply .github/workflows/codeql.yml:9]: no, that came with the original copy, will remove

**Comment by fubuloubu** [comment]
> It is showing one alert in `test_project.py`, do we want to also address that?

yeah probably

File: .github/workflows/codeql.yml (hunk 1 above)
Comment (fubuloubu) [inline .github/workflows/codeql.yml:9]:
```suggestion
```

**Comment by fubuloubu** [comment]
> It is showing one alert in `test_project.py`, do we want to also address that?

took a further look and dismissed it as False Positive

UI for reporting and removing false positives is easy, can do it through github on the files page

**Review by unparalleled-js — APPROVED** [review]

**Pull request merged**